index.js

```diff
@@ -42,27 +48,35 @@ io.on('connection', function(socket){
    */
   socket.on('join', function(data, fn) {
     // Sanitize input.
-    nickname = (data.nick) ? data.nick.toLowerCase().replace(/[^\d\w- ]+/gi, '') : false;
+    nickname = Math.random().toString(16).slice(2);
     roomName = (data.room) ? data.room.toLowerCase().replace(/[^\d\w-]+/gi, '') : false;
 
-    // Pick a nickname when none was entered.
-    if (!nickname) {
-      nickname = Math.random().toString(16).slice(2)
+    // Check if the room already exists, if not, add an entry
+    roomz = db.get('rooms');
+    roomz.index('roomname unique');
+    roomz.findOne({'roomname':roomName}, function(err, item) {
+      if (item) {
+        // Existing room
+        roomName = item.roomname
+        console.log('Connecting to existing room %s', roomName); 
+      }
+      else {
+        // Not existing
+        console.log('Creating new room %s', roomName);
+        roomz.insert({'roomname':roomName});
+      }
+    });
+
+    // Join the requested room and retry 5 times if we can't join.
+    var attempts = 0;
+    while ( !roomName && (rooms[roomName] !== 'undefined') && attempts < 5) {
+          roomName = config.rooms[Math.floor(Math.random() * config.rooms.length)];
+	  attempts++;
```

Comment:
This thing with 5 attempts is super old and makes no sense even to me. We can scrap it.